### ISSUE TITLE
Report error/warning if /pathmap doesn't cover all source paths

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#23969]
**Version Used**: 

2.7.0.62412 (84aeb8e6)

**Steps to Reproduce**:

1. 
Create files ```C:\temp\a\x.cs``` and ```C:\temp\b\y.cs``` containing some methods.

```C:\temp>csc a\x.cs b\y.cs /pathmap:C:\temp\a=/_/ /t:library /deterministic /debug:portable```
 
The resulting binary (pdb) is not fully deterministic since not all paths in the pdb were mapped to a deterministic value. 

**Expected Behavior**:

An error or warning is reported when ```/deterministic``` and ```/pathmap``` are specified but the map doesn't cover all paths, so that the user knows that the given ```/pathmap``` is insufficient.

**Actual Behavior**:

No warning/error.
